@ServerResponseFilter and @ServerRequestFilter fails on startup with NoSuchMethodException

### Describe the bug

When upgrading to Quarkus 2.14.10 startup of application with @ServerResponseFilter and @ServerRequestFilter crash on startup with
```
... 
Caused by: java.lang.NoSuchMethodException: org.acme.RequestLogger$GeneratedServerRequestFilter$filter.<init>()
	at java.base/java.lang.Class.getConstructor0(Class.java:3617)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2786)
	at io.quarkus.arc.runtime.BeanContainerImpl$DefaultInstanceFactory.create(BeanContainerImpl.java:78)
	... 26 more
```
(Or org.acme.RequestLogger$GeneratedServerResponseFilter$filter.<init>())

### Expected behavior

The filters are created and app starts successfully

### Actual behavior

Application crash on startup

### How to Reproduce?

https://github.com/computerlove/NoSuchMethodException-GeneratedServerRequestFilter
The application crash on startup when running `mvn quarkus:dev`

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

17

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.14.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/AnotherValidNonBlockingFiltersTest.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/AnotherValidNonBlockingFiltersTest.java
@@ -6,6 +6,7 @@ import java.io.IOException;
 import java.util.function.Supplier;
 
 import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Priorities;
@@ -151,6 +152,7 @@ public class AnotherValidNonBlockingFiltersTest {
         }
     }
 
+    @ApplicationScoped
     public static class CustomFilters {
 
         @ServerRequestFilter(nonBlocking = true, priority = Priorities.USER + 20)

--- a/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/ConditionalBeanFiltersTest.java
+++ b/extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/test/java/io/quarkus/resteasy/reactive/server/test/customproviders/ConditionalBeanFiltersTest.java
@@ -9,6 +9,7 @@ import java.util.Optional;
 import java.util.function.Supplier;
 import java.util.stream.Collectors;
 
+import javax.inject.Singleton;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Priorities;
@@ -134,6 +135,7 @@ public class ConditionalBeanFiltersTest {
         }
     }
 
+    @Singleton
     public static class AlwaysEnabledFilter {
 
         @ServerRequestFilter(priority = Priorities.USER + 100)

--- a/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java
+++ b/independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/generation/filters/CustomFilterGenerator.java
@@ -578,10 +578,10 @@ final class CustomFilterGenerator {
      */
     private ScopeInspectionResult inspectScope(ClassInfo classInfo) {
         if (classInfo.hasDeclaredAnnotation(ApplicationScoped.class) || classInfo.hasDeclaredAnnotation(RequestScoped.class)) {
-            return new ScopeInspectionResult(null, true);
+            return new ScopeInspectionResult(ApplicationScoped.class, true);
         }
         if (classInfo.hasDeclaredAnnotation(Singleton.class)) {
-            return new ScopeInspectionResult(null, false);
+            return new ScopeInspectionResult(Singleton.class, false);
         }
         List<FieldInfo> fields = classInfo.fields();
         if (fields.isEmpty()) {

